telos-backend/src/routes/api/journal.js

```diff
@@ -1,14 +1,119 @@
 const express = require('express');
+const CalendarEvent = require('../../models/calendar_event');
+const Habit = require('../../models/habit');
+const Text = require('../../models/text');
+const TodoTask = require('../../models/todo_task');
+const Widget = require('../../models/widget');
 
 const router = express.Router();
 
+// Helper methods ---------------------------------
+// Return a string representation of current day. 
+// See daysOfWeekenum in src/models/habit.js
+function getStringDayOfWeek(dayNum) {
+  let dayString;
+  switch (dayNum) {
+    case 0:
+      dayString = 'sun';
+      break;
+    case 1:
+      dayString = 'mon';
+      break;
+    case 2:
+      dayString = 'tue';
+      break;
+    case 3:
+      dayString = 'wed';
+      break;
+    case 4:
+      dayString = 'thu';
+      break;
+    case 5:
+      dayString = 'fri';
+      break;
+    case 6:
+      dayString = 'sat';
+      break;
+    default:
+      dayString = 'err';
+  }
+  return dayString;
+}
+
+async function getTodosForDay(date){
+  // Todos should return either what is made today, or all overdue tasks.
+  const todos = await TodoTask.find({
+    $or:[
+      { createdDate: { $gte: date + 'T00:00:00', $lte: date + 'T23:59:59' } }, 
+      { dueDate: {$gte: date + 'T00:00:00'} }
+    ] 
+  });
+  return todos;
+}
+
+// Param: date, of type date, parsed from url path param.
+async function getHabitsForDay(date){
+  // Create habit objects to return. These are not stored directly (since technically could be unlimited)
+  // Must find all habits that are between the requested day, but before end date, then check if they fall
+  // on the current day.
+  const habits = await Habit.find({ startDate: {$gte: date + 'T00:00:00'}, endDate: {$lte: date + 'T23:59:59'}});
```

Comment:
I think the startDate should be: 
$lte: date + 'T23:59:59'
and the endDate should be:
$gte: date + 'T00:00:00'

Basically, it's currently swapped around atm, since we want the provided date to be between the start and end dates.

Reply:
Yep good catch! Swapped around